Some compilers do not accept `.mp_obj_type` in 1.20

I'm finding that Emscripten's clang version is refusing to accept the way some structs are being initialized now. This doesn't happen before the 1.20 merge so I assume it's related to upstream changing how object types are defined. Any ideas?

```
build/lvgl/lv_mpy.c:1138:20: error: initializer element is not a compile-time constant
    .mp_obj_type = mp_lv_LV_RES_type_base,
                   ^~~~~~~~~~~~~~~~~~~~~~
build/lvgl/lv_mpy.c:1184:20: error: initializer element is not a compile-time constant
    .mp_obj_type = mp_lv_ENUM_LV_LOG_LEVEL_type_base,
                   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
build/lvgl/lv_mpy.c:1246:20: error: initializer element is not a compile-time constant
    .mp_obj_type = mp_lv_LV_ALIGN_type_base,
                   ^~~~~~~~~~~~~~~~~~~~~~~~
build/lvgl/lv_mpy.c:1294:20: error: initializer element is not a compile-time constant
    .mp_obj_type = mp_lv_LV_DIR_type_base,
```

<details>
<summary>Code near line 1246</summary>

```c
STATIC MP_DEFINE_CONST_OBJ_TYPE(
    mp_lv_LV_ALIGN_type_base,
    MP_QSTR_LV_ALIGN,
    MP_TYPE_FLAG_NONE,
    print, LV_ALIGN_print,
    
    
    attr, call_parent_methods,
    
    
    locals_dict, &LV_ALIGN_locals_dict
);

STATIC const mp_lv_obj_type_t mp_lv_LV_ALIGN_type = {
#ifdef LV_OBJ_T
    .lv_obj_class = NULL,
#endif
    .mp_obj_type = mp_lv_LV_ALIGN_type_base,
};
```
</details>

Seems to have moved the problem to a different location. Progress?

```
build/lvgl/lv_mpy.c:15952:13: error: initializer element is not a compile-time constant
 15952 |     parent, mp_lv_obj_type.mp_obj_type,
       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~
../../py/obj.h:783:250: note: expanded from macro 'MP_DEFINE_CONST_OBJ_TYPE'
  783 | #define MP_DEFINE_CONST_OBJ_TYPE(...) MP_DEFINE_CONST_OBJ_TYPE_EXPAND(MP_DEFINE_CONST_OBJ_TYPE_NARGS(__VA_ARGS__, _INV, 12, _INV, 11, _INV, 10, _INV, 9, _INV, 8, _INV, 7, _INV, 6, _INV, 5, _INV, 4, _INV, 3, _INV, 2, _INV, 1, _INV, 0)(mp_obj_type_t, __VA_ARGS__))
      |                                                                                                                                                                                                                                                          ^~~~~~~~~~~
../../py/obj.h:749:415: note: expanded from macro 'MP_DEFINE_CONST_OBJ_TYPE_NARGS_7'
  749 | #define MP_DEFINE_CONST_OBJ_TYPE_NARGS_7(_struct_type, _typename, _name, _flags, f1, v1, f2, v2, f3, v3, f4, v4, f5, v5, f6, v6, f7, v7) const _struct_type _typename = { .base = { &mp_type_type }, .name = _name, .flags = _flags, .slot_index_##f1 = 1, .slot_index_##f2 = 2, .slot_index_##f3 = 3, .slot_index_##f4 = 4, .slot_index_##f5 = 5, .slot_index_##f6 = 6, .slot_index_##f7 = 7, .slots = { v1, v2, v3, v4, v5, v6, v7, } }
      |                                                                                                                                                                                                                                                                                                                                                                                                                               ^~
../../py/obj.h:770:44: note: expanded from macro 'MP_DEFINE_CONST_OBJ_TYPE_EXPAND'
  770 | #define MP_DEFINE_CONST_OBJ_TYPE_EXPAND(x) x
      |                                            ^
```

```c
STATIC MP_DEFINE_CONST_OBJ_TYPE(
    mp_lv_canvas_type_base,
    MP_QSTR_canvas,
    MP_TYPE_FLAG_NONE,
    print, canvas_print,
    make_new, canvas_make_new,
    binary_op, mp_lv_obj_binary_op,
    attr, call_parent_methods,
    buffer, mp_lv_obj_get_buffer,
    parent, mp_lv_obj_type.mp_obj_type, // <---- this is the line that errors
    locals_dict, &canvas_locals_dict
);
```

Unfortunately looks about the same to me.
```
build/lvgl/lv_mpy.c:15429:34: error: initializer element is not a compile-time constant
 15429 |     { MP_ROM_QSTR(MP_QSTR_CTRL), MP_ROM_PTR(mp_lv_LV_BTNMATRIX_CTRL_type.mp_obj_type) },
       |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../py/obj.h:348:23: note: expanded from macro 'MP_ROM_PTR'
  348 | #define MP_ROM_PTR(p) (p)
      |                       ^~~
build/lvgl/lv_mpy.c:15462:13: error: initializer element is not a compile-time constant
 15462 |     parent, mp_lv_obj_type.mp_obj_type,
       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~
../../py/obj.h:783:250: note: expanded from macro 'MP_DEFINE_CONST_OBJ_TYPE'
  783 | #define MP_DEFINE_CONST_OBJ_TYPE(...) MP_DEFINE_CONST_OBJ_TYPE_EXPAND(MP_DEFINE_CONST_OBJ_TYPE_NARGS(__VA_ARGS__, _INV, 12, _INV, 11, _INV, 10, _INV, 9, _INV, 8, _INV, 7, _INV, 6, _INV, 5, _INV, 4, _INV, 3, _INV, 2, _INV, 1, _INV, 0)(mp_obj_type_t, __VA_ARGS__))
      |                                                                                                                                                                                                                                                          ^~~~~~~~~~~
../../py/obj.h:749:415: note: expanded from macro 'MP_DEFINE_CONST_OBJ_TYPE_NARGS_7'
  749 | #define MP_DEFINE_CONST_OBJ_TYPE_NARGS_7(_struct_type, _typename, _name, _flags, f1, v1, f2, v2, f3, v3, f4, v4, f5, v5, f6, v6, f7, v7) const _struct_type _typename = { .base = { &mp_type_type }, .name = _name, .flags = _flags, .slot_index_##f1 = 1, .slot_index_##f2 = 2, .slot_index_##f3 = 3, .slot_index_##f4 = 4, .slot_index_##f5 = 5, .slot_index_##f6 = 6, .slot_index_##f7 = 7, .slots = { v1, v2, v3, v4, v5, v6, v7, } }
      |                                                                                                                                                                                                                                                                                                                                                                                                                               ^~
../../py/obj.h:770:44: note: expanded from macro 'MP_DEFINE_CONST_OBJ_TYPE_EXPAND'
  770 | #define MP_DEFINE_CONST_OBJ_TYPE_EXPAND(x) x
      |                                            ^
build/lvgl/lv_mpy.c:15771:13: error: initializer element is not a compile-time constant
 15771 |     parent, mp_lv_obj_type.mp_obj_type,
       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

Much better, now only the `MP_ROM_PTR` one remains.

```
build/lvgl/lv_mpy.c:12731:34: error: initializer element is not a compile-time constant
 12731 |     { MP_ROM_QSTR(MP_QSTR_FLAG), MP_ROM_PTR(mp_lv_LV_OBJ_FLAG_type.mp_obj_type) },
       |                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../py/obj.h:348:23: note: expanded from macro 'MP_ROM_PTR'
  348 | #define MP_ROM_PTR(p) (p)
      |                       ^~~
```

--- a/gen/gen_mpy.py
+++ b/gen/gen_mpy.py
@@ -711,7 +711,7 @@ if len(obj_names) > 0:
 
 typedef struct mp_lv_obj_type_t {{
     const lv_obj_class_t *lv_obj_class;
-    mp_obj_type_t mp_obj_type;
+    const mp_obj_type_t *mp_obj_type;
 }} mp_lv_obj_type_t;
 
 STATIC const mp_lv_obj_type_t mp_lv_{base_obj}_type;
@@ -719,7 +719,7 @@ STATIC const mp_lv_obj_type_t *mp_lv_obj_types[];
 
 STATIC inline const mp_obj_type_t *get_BaseObj_type()
 {{
-    return &mp_lv_{base_obj}_type.mp_obj_type;
+    return mp_lv_{base_obj}_type.mp_obj_type;
 }}
 
 MP_DEFINE_EXCEPTION(LvReferenceError, Exception)
@@ -931,7 +931,7 @@ STATIC inline mp_obj_t lv_to_mp(LV_OBJ_T *lv_obj)
         const mp_lv_obj_type_t **iter = &mp_lv_obj_types[0];
         for (; *iter; iter++) {
             if ((*iter)->lv_obj_class == lv_obj_class) {
-                mp_obj_type = &(*iter)->mp_obj_type;
+                mp_obj_type = (*iter)->mp_obj_type;
                 break;
             }
         }
@@ -1033,7 +1033,7 @@ MP_REGISTER_ROOT_POINTER(void *mp_lv_user_data);
 #else // LV_OBJ_T
 
 typedef struct mp_lv_obj_type_t {
-    mp_obj_type_t mp_obj_type;
+    mp_obj_type_t *mp_obj_type;
 } mp_lv_obj_type_t;
 
 #endif
@@ -2555,7 +2555,7 @@ def gen_obj_methods(obj_name):
     obj_metadata[obj_name]['members'].update({get_enum_member_name(enum_member_name): {'type':'enum_member'} for enum_member_name in get_enum_members(obj_name)})
     # add enums that match object name
     obj_enums = [enum_name for enum_name in enums.keys() if is_method_of(enum_name, obj_name)]
-    enum_types = ["{{ MP_ROM_QSTR(MP_QSTR_{name}), MP_ROM_PTR(&mp_lv_{enum}_type.mp_obj_type) }}".
+    enum_types = ["{{ MP_ROM_QSTR(MP_QSTR_{name}), MP_ROM_PTR(&mp_lv_{enum}_type_base) }}".
                     format(name=sanitize(method_name_from_func_name(enum_name)), enum=enum_name) for enum_name in obj_enums]
     obj_metadata[obj_name]['members'].update({method_name_from_func_name(enum_name): {'type':'enum_type'} for enum_name in obj_enums})
     for enum_name in obj_enums:
@@ -2633,11 +2633,11 @@ STATIC MP_DEFINE_CONST_OBJ_TYPE(
     locals_dict, &{obj}_locals_dict
 );
 
-STATIC const mp_lv_obj_type_t mp_lv_{obj}_type = {{
+GENMPY_UNUSED STATIC const mp_lv_obj_type_t mp_lv_{obj}_type = {{
 #ifdef LV_OBJ_T
     .lv_obj_class = {lv_class},
 #endif
-    .mp_obj_type = mp_lv_{obj}_type_base,
+    .mp_obj_type = &mp_lv_{obj}_type_base,
 }};
     """.format(
             module_name = module_name,
@@ -2648,7 +2648,7 @@ STATIC const mp_lv_obj_type_t mp_lv_{obj}_type = {{
             make_new = 'make_new, %s_make_new,' % obj_name if is_obj else '',
             binary_op = 'binary_op, mp_lv_obj_binary_op,' if is_obj else '',
             buffer = 'buffer, mp_lv_obj_get_buffer,' if is_obj else '',
-            parent = 'parent, &mp_lv_%s_type.mp_obj_type,' % parent_obj_names[obj_name] if obj_name in parent_obj_names and parent_obj_names[obj_name] else '',
+            parent = 'parent, &mp_lv_%s_type_base,' % parent_obj_names[obj_name] if obj_name in parent_obj_names and parent_obj_names[obj_name] else '',
             lv_class = '&lv_%s_class' % obj_name if is_obj else 'NULL',
             ))
 
@@ -2888,11 +2888,11 @@ STATIC const mp_rom_map_elem_t {module_name}_globals_table[] = {{
 }};
 """.format(
         module_name = sanitize(module_name),
-        objects = ''.join(['{{ MP_ROM_QSTR(MP_QSTR_{obj}), MP_ROM_PTR(&mp_lv_{obj}_type.mp_obj_type) }},\n    '.
+        objects = ''.join(['{{ MP_ROM_QSTR(MP_QSTR_{obj}), MP_ROM_PTR(&mp_lv_{obj}_type_base) }},\n    '.
             format(obj = sanitize(o)) for o in obj_names]),
         functions =  ''.join(['{{ MP_ROM_QSTR(MP_QSTR_{name}), MP_ROM_PTR(&mp_{func}_mpobj) }},\n    '.
             format(name = sanitize(simplify_identifier(f.name)), func = f.name) for f in module_funcs]),
-        enums = ''.join(['{{ MP_ROM_QSTR(MP_QSTR_{name}), MP_ROM_PTR(&mp_lv_{enum}_type.mp_obj_type) }},\n    '.
+        enums = ''.join(['{{ MP_ROM_QSTR(MP_QSTR_{name}), MP_ROM_PTR(&mp_lv_{enum}_type_base) }},\n    '.
             format(name = sanitize(get_enum_name(enum_name)), enum=enum_name) for enum_name in enums.keys() if enum_name not in enum_referenced]),
         structs = ''.join(['{{ MP_ROM_QSTR(MP_QSTR_{name}), MP_ROM_PTR(&mp_{struct_name}_type) }},\n    '.
             format(name = sanitize(simplify_identifier(struct_name)), struct_name = sanitize(struct_name)) for struct_name in generated_structs \

